core/eolearn/core/eodata_io.py

```diff
@@ -95,6 +120,17 @@ def save_eopatch(
     else:
         _check_letter_case_collisions(eopatch_features, [])
 
+
+def _remove_old_eopatch(filesystem: FS, patch_location: str) -> None:
+    filesystem.removetree(patch_location)
+    if patch_location != "/":  # avoid redundant filesystem.makedirs if the location is '/'
+        filesystem.makedirs(patch_location, recreate=True)
```

Comment:
Can you explain this in more detail? I'm not sure what exactly is the reasoning here (I know there's no real change from before, just trying to follow the code).

Reply:
if someone does `EOPatch().save("/", filesystem)` the root is never removed so it doesnt need to be recreated (or so Matej A claims).

I do feel like this is redundant (especially since `OVERWRITE_PATCH` is not a common choice), but didn't want to alter code in a refactor MR. But since you pointed it out, we might as well remove the conditional if we all agree :)

Reply:
Ah I see yes in [the docs](/Users/mlubej/work/projects/sh-project/eo-learn/example_data/TestEOPatch')

So it's just optimizing to not call it, since the code wouldn't actually remove it, but the question is what do we really save anyway. I also think we can remove the conditional.

Reply:
it's saving a single IO call in a very very niche case. I agree that it's not worth the code complication. Removed